glossary.md

```diff
@@ -0,0 +1,37 @@
+A Glossary of Stateless Ethereum
+---------------------------------
+*This glossary is a work in progress; contributions are welcome*
+
+**Beam Sync**. An alternative sync protocol developed by the Trinity team with similarities to what is imagined for Stateless Ethereum. The scheme works by executing a block first, and then requesting missing state data on demand in the form of a witness.
+
+**Code Merkleization**. A method of splitting contract bytecode into smaller pieces, so that an execution that only touches part of a smart contract does not need to include the entire contract in its witness.
+
+**Merkle Tree**. A [tree](https://en.wikipedia.org/wiki/Tree_(data_structure)) data structure in which every leaf node is labelled with the [hash](https://en.wikipedia.org/wiki/Cryptographic_hash_function) of a data block, and every non-leaf node is labelled with the [cryptographic hash](https://en.wikipedia.org/wiki/Cryptographic_hash_function) of the labels of its child nodes.
```

Comment:
both the hash and the cryptographic hash is the same algorithm in case of Ethereum, so I think they all should be names the same. otherwise, it might look like they are different for some reason.